JavaScript/JavaScript.sublime-syntax

```diff
@@ -914,83 +902,65 @@ contexts:
 
   either-function-declaration:
     - match: '(?={{func_lookahead}})'
-      push: function-declaration
+      set: function-declaration
     - match: '(?={{arrow_func_lookahead}})'
-      push: arrow-function-declaration
+      set: arrow-function-declaration
 
   function-declaration:
-    - meta_scope: meta.function.declaration.js
-    - match: '\b(async)\b\s*'
-      captures:
-        1: storage.type.js
-    - match: \b(function)\b(\*)?\s*
-      captures:
-        1: storage.type.function.js
-        2: keyword.generator.asterisk.js
-    - match: '{{identifier}}'
-      scope: entity.name.function.js
-    - include: function-declaration-parameters
-    - match: '(?=\s*\{)'
-      set: function-block
+    - match: ''
+      set:
+        - - include: immediately-pop
+        - function-block
```

Comment:
Since the `function-block` context currently has a `meta_scope` of `meta.block.js`, doesn't setting it into this list mean that everything set after it (so to speak) will get the `meta.block.js` scope? (e.g. if I have `function a() {}`, all of it will be scoped with `meta.block.js`. Perhaps the `function-block` scope should be reverted to applying the meta scope on the pushed context, and then on the ending `}` again?